### PR TITLE
Buffer data removal

### PR DESCRIPTION
**Pull request by Pessimistress**
For https://github.com/uber/deck.gl/issues/1974

#### Change List
- Remove reference to `buffer.data`

### CODE REVIEW

File: modules/core/src/lib/attribute-transition-utils.js [hunk 1]
```diff
@@ -107,3 +108,10 @@ export function padBuffer({
 
   fromState.setData({data});
 }
+
+export function getBufferLength(buffer) {
+  if (buffer.data) {
```
Comment (1chandu) [inline modules/core/src/lib/attribute-transition-utils.js]: Is this temporary? wouldn't this always be false once luma PR lands?

Comment (Pessimistress) [reply modules/core/src/lib/attribute-transition-utils.js]: Doesn't it keep some data for debug purpose? If not, can we store the element length in the Buffer object?

Comment (1chandu) [reply modules/core/src/lib/attribute-transition-utils.js]: Yes we will be adding getLength support to Buffer eventually, it is going to be part of this PR : https://github.com/uber/luma.gl/pull/423

Comment (1chandu) [reply modules/core/src/lib/attribute-transition-utils.js]: if all you need is 'BYTES_PER_ELEMENT', can you use `getTypedArrayFromGLType` + buffer.accessor.type ? We can clean this up once we have full support in luma.gl

Comment (1chandu) [reply modules/core/src/lib/attribute-transition-utils.js]: Added support in luma.gl https://github.com/uber/luma.gl/pull/592

**Pull request merged**